content/blog/migrating-vsphere-virtual-volumes-to-kubevirt-with-hpe-csi-driver-for-kubernetes.md

```diff
@@ -0,0 +1,432 @@
+    args:
+      creates: "{{ var_lib_mysql }}/employees"
+      chdir: "{{ ansible_user_dir }}/test_db"
+
+  - name: Test Test_DB
+    shell: mysql -t < test_employees_sha.sql && mysql -t <<< 'select @@hostname as 'VM';'
+    args:
+      chdir: "{{ ansible_user_dir }}/test_db"
+    register: test_db_results
+
+  - name: Show results
+    pause:
+      prompt: "{{ test_db_results.stdout }}"
+      seconds: 10
+```
+
+Another detail in the playbook that might need clarification is that the Test_DB sources is downloaded to the destination VM only because the database queries are needed for validation of the cloning and migration.
+
+The last few output lines of the playbook reveal that the database was deployed and tested properly.
+
+```shell
+...
++---------+--------+
+| summary | result |
++---------+--------+
+| CRC     | OK     |
+| count   | OK     |
++---------+--------+
++----------------+
+| VM             |
++----------------+
+| mariadb-legacy |
++----------------+:
+ok: [mariadb-legacy]
+
+PLAY RECAP *************************************************************************
+mariadb-kubevirt           : ok=6    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
+mariadb-legacy             : ok=15   changed=2    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
+```
+
+Once the prerequisites have been installed, we can also inspect the hypervisor platform each VM is running on, to ensure what we’re doing is actually being executed where we expect it.
+
+```shell
+$ ansible -m command all -b -a virt-what
+mariadb-legacy | CHANGED | rc=0 >>
+vmware
+mariadb-kubevirt | CHANGED | rc=0 >>
+redhat
+kvm
+```
+
+# Preparing for the first failure
+
+What is being described in this blog is a perfectly canned example. In reality, the first experiment always fails. Maybe even the second, and third. The point is that it should be non-disruptive and easy to recover from the failures to iteratively improve as the migration is taken across the finish line with as short downtime as possible. What happens before the cutover may take days and weeks of planning and risk free test execution.
+
+In the first iteration, the vSphere vVol needs to be cloned into a PersistentVolume (PV) using a Persistent Volume Claim (PVC). The PVC is then attached to the KubeVirt VM as a disk.
+
+PVC creation may rely on pre-provisioned “static” PVs but more commonly a StorageClass is used to dynamically provision the PV. To allow the PV to clone a pree-xisting volume on the storage array the HPE CSI Driver for Kubernetes needs to be supplied a parameter with the corresponding volume name.
+
+The CSI driver supports a construct that allows Kubernetes users to supply their own values to StorageClass parameters called “allowOverrides”. In this exercise we need to craft a StorageClass that allows users to influence the “importVolumeName” (used for migration) and “importVolAsClone” (used for cloning).
```

Comment:
Modify the second sentence to read:

**"For** this exercise, **you'll** need to craft a StorageClass..."